mocktracer/mockspan.go

```diff
@@ -183,8 +185,26 @@ func (s *MockSpan) Finish() {
 func (s *MockSpan) FinishWithOptions(opts opentracing.FinishOptions) {
 	s.Lock()
 	s.FinishTime = opts.FinishTime
-	s.logs = append(s.logs, opts.BulkLogData...)
 	s.Unlock()
+
+	// Handle any late-bound LogRecords.
+	for _, lr := range opts.LogRecords {
+		s.logFieldsWithTimestamp(lr.Timestamp, lr.Fields...)
+	}
+	// Handle (deprecated) BulkLogData.
```

Comment:
this code gives a sense of what Tracer implementations might do to support the old API during a migration period